### PR TITLE
[Fix] Removed unnecessary IIFEs from ES6 challenges

### PR DESCRIPTION
**Pull request by ezioda004**
Removed IIFE's, added solution and test to one of the challenges.
Successfully tested locally.

- [x] I have read [freeCodeCamp's contribution guidelines](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/CONTRIBUTING.md).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `master` branch of freeCodeCamp.
- [x] None of my changes are plagiarized from another source without proper attribution.
- [x] My article does not contain shortened URLs or affiliate links.

Closes #XXXXX

### CODE REVIEW

**Comment by ezioda004** [comment]
@ValeraS Can you check the build error? The solution should pass the CI, unless I'm missing something.

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/es6/use-destructuring-assignment-to-pass-an-object-as-a-functions-parameters.english.md
Comment (ValeraS) [inline curriculum/challenges/english/02-javascript-algorithms-and-data-structures/es6/use-destructuring-assignment-to-pass-an-object-as-a-functions-parameters.english.md]: @ezioda004 solution should contain a seed code.

Comment (ezioda004) [reply curriculum/challenges/english/02-javascript-algorithms-and-data-structures/es6/use-destructuring-assignment-to-pass-an-object-as-a-functions-parameters.english.md]: Ahh fixed, thanks!

**Review by Nirajn2311 — APPROVED** [review]
@ezioda004 Just tested it locally, everything is working perfectly 🎉 🎉

**Comment by RandellDawson** [comment]
@ezioda004 Your last commit show 490 files changed.  Did you do a rebase before merging?

image

**Comment by ezioda004** [comment]
I fixed a merge conflict, I guess github evened `fix-es6` branch with `fcc:master` in the process.

**Comment by scissorsneedfoodtoo** [comment]
So sorry for the delay in reviewing this, @ezioda004. Missed this one in my feed! Hope you don't mind, but I took care of the merge conflicts and made a minor change to the spacing in one of the challenges. Tested the challenges and everything LGTM, but will ask for another review before we merge this.

**Comment by moT01** [comment]
@ezioda004 @scissorsneedfoodtoo I tested locally, found one small issue(I think)...
In `use-destructuring-assignment-to-pass-an-object-as-a-functions-parameters`
The new test added for `destructured parameter was used`
Doesn’t seem to work quite right

This code…
```
// change code below this line
function half(stats) {
// use function argument destructuring
  const {max, min} = stats;
  return (max + min) / 2.0;
}
// change code above this line
```
Causes that test to pass - it’s supposed to fail, I believe, since it was not destructured in the parameter/argument. 

However, it does cause the `Destructuring was used` test to fail (the correct code passes both). I didn't look into the details of the regex, but it seems like the titles of the two tests are just reversed. So perhaps we could reverse the titles/error text (whatever you want to call it) of the tests - or maybe remove the new test and change the title of the old one to `destructured parameter was used` or `destructuring was used in the parameter`. 

Other than that, everything else looked good. Nice job @ezioda004.

**Comment by RandellDawson** [comment]
@ezioda004 The only change I might suggest would be to use an arrow functions and `const` and `let` in the Challenge Seeds, because all of these have already been introduced in the beginning of the ES6 section. 

I noticed the original versions used these features, so I was curious why you did not do the same.

**Comment by thecodingaviator** [comment]
@ezioda004 Any progress?

**Comment by RandellDawson** [comment]
OK, I have made the following changes to this PR:

1.  Fixed the outstanding merge conflicts

2. Removed the assert message argument from all the tests

3. Made all the function declarations use const syntax

4. Corrected (hopefully) the issues noted by @moT01 

5 Resolved the hack described in [this forum post](https://www.freecodecamp.org/forum/t/js-use-destructuring-assignment-to-pass-an-object-as-a-functions-parameters/266368/4)

Let me know if I missed anything.  I just want to get this PR merged as it is holding up other PRs from being merged at the moment.

**Comment by RandellDawson** [comment]
@thecodingaviator, @Manish-Giri, @moT01 Can we get two reviews on this one?  Other PRs can be merged once this one is merged.

**Review by thecodingaviator — APPROVED** [review]
LGTM! I've tested all challenges locally! 👍

**Review by moT01 — APPROVED** [review]
This LGTM @ezioda004, thanks for all the work on this everyone. I still have some concerns about some of the instructions being a little confusing and some more things that could be better, but I think they are outside the scope of this PR - all these changes look good.

**Pull request merged**
